src/qililab/qprogram/qblox_compiler.py

```diff
@@ -341,15 +342,34 @@ def _handle_wait(self, element: Wait):
         if isinstance(element.duration, Variable):
             duration = self._buses[element.bus].variable_to_register[element.duration]
             self._buses[element.bus].dynamic_durations.append(element.duration)
+            self._buses[element.bus].qpy_block_stack[-1].append_component(
+                component=QPyInstructions.Wait(wait_time=duration)
+            )
```

Comment:
What if the variable goes over INST_MAX_WAIT? Shouldn't we handle it the same way?

```
with qp.for_loop(variable=duration, start=100, stop=2 * INST_MAX_WAIT, step=1000):
    qp.wait(duration=duration)
```

Reply:
Yes I mention this in the comment above

> I removed I line I had previously added raising an error if the duration of a dynamic wait was longer than qblox INST_MAX_WAIT because the value of the wait is not known until allocated after compilation. I think this is ok however, since the wait instruction from qpysequence already raises an error if a duration longer than INST_MAX_WAIT is allocated (so an error expliciting what went wrong is already in place)

QPysequence already raises an error for this. Would you do it otherwise?

Reply:
Oh I didn't see the comment.

Well, in general, if we can catch an error in Qililab level it's better. Since we can provide our own meaningful error etc. And user's shouldn't actually know or care what is going on behind the scenes (for example that we use qpysequence).